core/mixins/_media-queries.scss

```diff
@@ -46,6 +46,14 @@
    @include respond-range(lap) {
      .foo {background: red;}
    }
+
+   @include respond-retina(2) {
+     .foo {background-size: 100px auto;}
+   }
+
+  @include respond-retina(144dpi) {
+     .foo {background-size: 100px auto;}
+   }
```

Comment:
I want to beef up this intro block now that there is a 3rd media query. 

---

Setup media query mixins for the following Media features (http://www.w3.org/TR/2012/REC-css3-mediaqueries-20120619/#media1):

- Width
- Height
- Resolution

The width and height mixins are for applying styles based on the width or height of the viewport, this can either be a minimum width or height or a maximum width or height, or a minimum width or height range or a maximum width or height range. The defaults are “width” and “minimum”. They accept any integer or any of Scally's breakpoints defined here: Core -> Settings -> Breakpoints. E.g.

```scss
@include respond-to(400) {
  .foo {background: red;}
}
```

Would compile to:

```scss
@media min-width(25em) {
  .foo {background: red;}
}
```

And for a range:

```scss
@include respond-range(500, 780) {
  .foo {background: red;}
}
```

Would compile to:

```scss
@media (min-width: 31.25em) and (max-width: 48.75em) {
  .foo {background: red;}
}
```

To apply a maximum width or height the "max" flag needs to be applied e.g.

```scss
@include respond-to(desk-large, max) {
  .foo {background: red;}
}
```

And for a range:

```scss
@include respond-range(500, 700, max) {
  .foo {background: red;}
}
```

And to target the height the `$axis` parameter needs to be set to "height" e.g.

```scss
@include respond-to(780, $axis: height) {
  .foo {background: red;}
}
```

And for a range:

```scss
@include respond-range(500, 700, $axis: height) {
  .foo {background: red;}
}
```

Applying both maximum and height will look something like this:  

```scss
@include respond-to(palm, max, $axis: height) {
  .foo {background: red;}
}
```

For a range it doesn’t make sense to pass in the “max” flag as ranges will always use a maximum width or height for the second parameter as ranges always need to be minimum–maximum however if the second parameter is omitted and the first parameter is a Scally breakpoint then the second parameter will use the next breakpoint in the sequence. So if you pass in the “lap” breakpoint for the first parameter and nothing else the second parameter will be “lap-large” as “lap-large” comes after “lap”. E.g. 

```scss
@include respond-range(lap, $axis: height) {
  .foo {background: red;}
}
```

Would compile to:

```scss
@media (min-height: 40.0625em) and (max-height: 75em) {
  .foo {background: red;}
}
```

Where `max-height: 75em` is the “lap-large” breakpoint.

N.B. the width and height mixins accept unitless integers or integers with `px` e.g.

```scss
@include respond-to(400) {
  .foo {background: red;}
}

@include respond-to(400px) {
  .foo {background: red;}
}
```

The resolution mixin is for applying styles to hi-dpi (retina) devices. It uses a default resolution defined here: Core -> Settings -> Retina Resolution or it will accept either a unitless device pixel ratio (dpr) e.g.

```scss
@include respond-retina(2) {
  .foo {background-size: 100px auto;}
}
```

Which will compile to a `dpi` unit using the calculation: “dpr * 96dpi” meaning the above mixin will compile too:

```scss
@media (min-resolution: 192dpi) {
  .foo {background-size: 100px auto;}
}
```

Or it will accept a `dpi` unit e.g.

```scss
@media (min-resolution: 192dpi) {
  .foo {background-size: 100px auto;}
}
```
---

If you don't want to add this np I'll add it when I merge this PR. The `@example`s can go too.